main.go

```diff
@@ -1,7 +1,328 @@
+	users = append(users, user)
+	//jwt?
+	//json.NewEncoder(w).Encode("")
+}
+
+func createTransfer(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+	params := mux.Vars(r)
+	for _, item := range certificates {
+		if item.ID == params["id"] && item.OwnerID == params["userId"] {
+			var certificate Certificate
+			//certificates = append(certificates[:index], certificates[index+1:]...)
+			err := json.NewDecoder(r.Body).Decode(&certificate)
+			if err != nil {
+				fmt.Fprintf(w, err.Error())
+				return
+			}
+			item.Transfer.To = certificate.Transfer.To
+			item.Transfer.Status = "Pending"
+		}
+	}
+}
+
+func acceptTransfer(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+	params := mux.Vars(r)
+	for _, item := range certificates {
+		if item.ID == params["id"] {
+			err := json.NewEncoder(w).Encode(item)
+			if err != nil {
+				fmt.Fprintf(w, err.Error())
+				return
+			}
+			item.OwnerID = params["userId"]
+			item.Transfer.To = ""
+			item.Transfer.Status = ""
+			break
+		}
+	}
+
+}
+
+//JWT generator
+func generateJWT() (string, error) {
+	fmt.Println("generateJWT...")
+	//Create the token
+	token := jwt.New(jwt.SigningMethodHS256)
+
+	//Create a map to store our claims
+	claims := token.Claims.(jwt.MapClaims)
+
+	claims["admin"] = true
+	claims["name"] = "Rob Wind"
+	claims["exp"] = time.Now().Add(time.Hour * 24).Unix()
+
+	//Sign the token
+	tokenString, err := token.SignedString(mySigningKey)
+
+	if err != nil {
+		fmt.Printf("Error: %s", err.Error())
+		return "", err
+	}
+
+	return tokenString, nil
+}
+
+//Get Token Handler
+var getTokenHandler = http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+	validToken, err := generateJWT()
+	if err != nil {
+		fmt.Fprintf(w, err.Error())
+	}
+
+	// fmt.Fprintf(w, validToken)
+	err = json.NewEncoder(w).Encode(validToken)
+	if err != nil {
+		fmt.Fprintf(w, err.Error())
+	}
+	//Write token to responsewriter
+	//json.NewEncoder(w).Encode(tokenString)
+	//w.Write([]byte(tokenString))
+})
+
+// Main
 func main() {
-	fmt.Println("Hello World")
+	fmt.Println("Running...")
+
+	initInMemoryData()
+
+	//Router
+	// Need to change routes to those 3 endpoints listed on the pdf.
+	r := mux.NewRouter()
+	r.HandleFunc("/", homeHandler).Methods("GET")
+	r.HandleFunc("/certificate", createCertificate).Methods("POST")
+	r.HandleFunc("/certificates", getCertificates).Methods("GET")
+	r.HandleFunc("/certificate/{id}", getCertificate).Methods("GET")
+	r.HandleFunc("/certificate/{id}", putCertificate).Methods("PUT")
+	r.HandleFunc("/certificate/{id}", patchCertificate).Methods("PATCH")
+	r.HandleFunc("/certificate/{id}", deleteCertificate).Methods("DELETE")
+	r.HandleFunc("/users/{userId}/certificates", getUsersCertificates).Methods("GET")
+	r.HandleFunc("/users/{userId}/certificates/{id}/transfers", createTransfer).Methods("POST") // PUT OR PATCH?
```

Comment:
Just posting data, we don't need to allow people to amend transfer requests. Maybe deleting a transfer / rejecting is needed. But they didn't ask for it. So that's something you can talk to them about.